PC/launcher.c

```diff
@@ -1040,32 +1046,47 @@ find_terminator(char * buffer, int len, BOM *bom)
 static BOOL
 validate_version(wchar_t * p)
 {
-    BOOL result = TRUE;
-
-    if (!isdigit(*p))               /* expect major version */
-        result = FALSE;
-    else if (*++p) {                /* more to do */
-        if (*p != L'.')             /* major/minor separator */
+    /*
+    Version information should start with one of 2 or 3,
+    Optionally followed by a period and a minor version,
+    Optionally followed by a minus and one of 32 or 64.
+    Valid examples:
+      2
+      3
+      2.7
+      3.6
+      2.7-32
+      The intent is to add to the valid patterns:
+      3.10
+      3-32
+      3.6-64
+      3-64
+    */
+    BOOL result = (p != NULL) && *p &&!iswdigit(*p);  /* Default to false if empty string or null pointer. */
```

Comment:
`result` should be true if the first character is a digit, and false for a `NULL` pointer or empty string, e.g. `result = (p != NULL) && iswdigit(*p)`. You don't need to check `*p` separately since `iswdigit` is false for a nul character, but it doesn't hurt if you want to keep that in. I'm just not understanding why you would negate the value of `iswdigit(*p)`.